### PR TITLE
Feature/flutter audio frame observer

### PR DESCRIPTION
**Pull request by jarekb123**
Implement streams for observing recorded/playback audio frames from Agora SDK

https://docs.agora.io/en/Video/API%20Reference/java/interfaceio_1_1agora_1_1rtc_1_1_i_audio_frame_observer.html#a386c3a451ce95f796d64e83006d7779c

is used

### CODE REVIEW

**Review by MlecznyU — APPROVED** [review]

File: android/src/main/kotlin/io/agora/agora_rtc_engine/FlutterAudioFrameObserver.kt [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+  val value = get(key)
+  return if (value is T?) {
+    value as T
+  } else {
+    default
+  }
+}
+
+class FlutterAudioFrameObserver(private val rtcEnginePlugin: AgoraRtcEnginePlugin) : IAudioFrameObserver {
+
```
Comment (gocal) [inline android/src/main/kotlin/io/agora/agora_rtc_engine/FlutterAudioFrameObserver.kt:41]: redundant space

File: android/src/main/kotlin/io/agora/agora_rtc_engine/FlutterAudioFrameObserver.kt [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+
+  val onRecordFrameStreamHandler = object : EventChannel.StreamHandler {
+    override fun onListen(arguments: Any?, events: EventChannel.EventSink?) {
+      var sampleRate = 48000
```
Comment (gocal) [inline android/src/main/kotlin/io/agora/agora_rtc_engine/FlutterAudioFrameObserver.kt:45]: constants ?

File: lib/rtc_audio_observer.dart [hunk 3]
```diff
@@ -0,0 +1,98 @@
+import 'dart:typed_data';
```
Comment (gocal) [inline lib/rtc_audio_observer.dart:1]: I'd rather move it to src/* and export it in rtc_engine

Comment (jarekb123) [reply lib/rtc_audio_observer.dart:1]: Ok, but imo the god class approach is really poor idea and breaking SRP

**Pull request merged**